### ISSUE TITLE
.setPauseAtEndOfMediaItems(true) not working

### ISSUE THREAD

**Issue by MrBzik** [brookmg/ExoRecord#3]
Hi. In general everything seems to be working fine. But I noticed option to make pause does not take effect now. I use hilt for instances like this : 

`    @Provides
    @ServiceScoped
    fun providesAudioAttributes() = AudioAttributes.Builder()
        .setContentType(C.AUDIO_CONTENT_TYPE_MUSIC)
        .setUsage(C.USAGE_MEDIA)
        .build()

    @Provides
    @ServiceScoped
    fun providesExoPlayer (
        @ApplicationContext app : Context,
        audioAttributes: AudioAttributes,
        renderersFactory: DefaultRenderersFactory
    ) = ExoPlayer.Builder(app, renderersFactory)
        .setAudioAttributes(audioAttributes, true)
        .setPauseAtEndOfMediaItems(true)
        .setHandleAudioBecomingNoisy(true)
        .build()

    @Provides
    @ServiceScoped
    fun providesDataSourceFactory (
        @ApplicationContext app : Context
    ) = DefaultDataSource.Factory(app)

    @Provides
    @ServiceScoped
    fun providesRendersFactory(
        @ApplicationContext app : Context,
        exoRecord: ExoRecord
    ) = object : DefaultRenderersFactory(app){
        override fun buildAudioSink(
            context: Context,
            enableFloatOutput: Boolean,
            enableAudioTrackPlaybackParams: Boolean,
            enableOffload: Boolean
        ): AudioSink? {
            return DefaultAudioSink.Builder()
                .setAudioCapabilities(DEFAULT_AUDIO_CAPABILITIES)
                .setAudioProcessorChain(DefaultAudioSink
                    .DefaultAudioProcessorChain(exoRecord.exoRecordProcessor))
                .setEnableFloatOutput(enableFloatOutput)
                .setEnableAudioTrackPlaybackParams(enableAudioTrackPlaybackParams)
                .build()

        }
    }

    @Provides
    @ServiceScoped
    fun providesExoRecord(@ApplicationContext app : Context) =
        ExoRecord(app.applicationContext as Application)`

So playbackState Player.STATE_ENDED also stopping to be triggered. 
When I remove    ".setAudioProcessorChain(DefaultAudioSink
                    .DefaultAudioProcessorChain(exoRecord.exoRecordProcessor))" it is working. 

Maybe I set It wrong? Or there is a workaround

**Comment by brookmg** [comment]
Hey @MrBzik, can you share a small project that can reproduce this?

**Comment by MrBzik** [comment]
Hi. Okay, I will get to it as soon as possible

**Comment by MrBzik** [comment]
So I made a project. Guess I just need to put link here.

https://github.com/MrBzik/TestingProject

It covers the described issue. When playing recording with initial player, it starts a loop, and logs isPlaying state. It is always true.
When switching to player without default renders factory, it sets isPlaying state to false as soon as file ends. 

Also, it covers another issue. The speed of recording is very fast. I encountered this with one of the streams (not the others), and included it  as a source. Don't know what is the cause for that. 

Noticed, that second issue is present in original wav file also, and seems to be exactly double speed. It happens with streams like podcasts, talking.

**Comment by MrBzik** [comment]
Anyway, I did not know I could pass valid data to processor before starting recording, like this : 
`exoRecord.exoRecordProcessor.configure(AudioProcessor.AudioFormat(sampleRate, channels,  C.ENCODING_PCM_16BIT))`
Seems to do the trick. 
As for playback stop guess I ll just go with for loop, comparing duration and progress. 
So don't bother too much. Thank you.
